plugins/gpg-agent/gpg-agent.plugin.zsh

```diff
@@ -1,14 +1,17 @@
-# Enable gpg-agent if it is not running
-GPG_AGENT_SOCKET="${XDG_RUNTIME_DIR}/gnupg/S.gpg-agent.ssh"
-if [ ! -S $GPG_AGENT_SOCKET ]; then
-  gpg-agent --daemon >/dev/null 2>&1
-  export GPG_TTY=$(tty)
+# Enable gpg-agent if it is not running-
+# --use-standard-socket will work from version 2 upwards
+
+AGENT_SOCK=`gpgconf --list-dirs | grep agent-socket | cut -d : -f 2`
+
+if [ ! -S ${AGENT_SOCK} ]; then
+  gpg-agent --daemon --use-standard-socket >/dev/null 2>&1
```

Comment:
--use-standard-socket since GnuPG 2.1 is default behaviour. And this option is not used.

Reply:
Again as you said, it would be nice not to need this, but it is necessary on 2.0

Reply:
Okay, but on 2.1 it can throw information about deprecation on each new session, so again we need to add version checking.

Reply:
I did wonder about this - but checking/parsing for the version felt error prone and complicated versus dumping the output to /dev/null which the plugin was alreading doing and  which will hide the  deprecation  warning?

Reply:
Version checking would give us also option to turn on support for gpg v1 again. Question is, if we want to support old GPG?
I think we shouldn't suppress messages for thing which is not generating any output if behaves correctly.
About cuts, I've checked behaviour of gpgconf on v2.0 and before 2.1.13 and now I understand why you used it. But we could use sth like ``` AGENT_SOCK=`gpgconf --list-dirs | grep agent-socket | cut -d : -f 2`'.ssh' ``` to do only one allocation.

I'll sit today/tomorrow and try to create some clean solution without hacks.

Reply:
> Version checking would give us also option to turn on support for gpg v1 again. Question is, if we > want to support old GPG?

The oldest box I use regularly is on CentIOS 6 - which was released in 2011 and that is using version 2.0.14 of gpg-agent 

The  --use-standard-socket option was introduced in: GnuPG 1.9.14 released (2004-12-22)

> I think we shouldn't suppress messages for thing which is not generating any output if behaves correctly.
Agree Completly. 

>About cuts, I've checked behaviour of gpgconf on v2.0 and before 2.1.13 and now I understand why you used it. But we could use sth like AGENT_SOCK=`gpgconf --list-dirs | grep agent-socket | cut -d : -f 2`'.ssh' to do only one allocation.

For the moment I've removed the allocation of GPG_SSH_AUTH_SOCK

> I'll sit today/tomorrow and try to create some clean solution without hacks.

Sounds good to me.